File: code-coaches/medium/name-buddy/name_buddy.rb
def contains_name_buddy(names, my_name)
  names.any? { |name| name[0] == my_name[0] }
end

def main
  names = gets.chomp.split
  my_name = gets.chomp

  if contains_name_buddy(names, my_name)
    puts "Compare notes"
  else
    puts "No such luck"
  end
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/name-buddy/name_buddy.py
def contains_name_buddy(names: list[str], my_name: str) -> bool:
    return any(name[0] == my_name[0] for name in names)


def main():
    names = input().split()
    my_name = input()

    if contains_name_buddy(names, my_name):
        print("Compare notes")
    else:
        print("No such luck")


if __name__ == "__main__":
    main()
